cmd/shipperctl/cmd/chart/app.go

```diff
@@ -0,0 +1,88 @@
+package chart
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/spf13/cobra"
+	"sigs.k8s.io/yaml"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/bookingcom/shipper/cmd/shipperctl/config"
+	shipper "github.com/bookingcom/shipper/pkg/apis/shipper/v1alpha1"
+)
+
+var (
+	appName  string
+	fileName string
+
+	renderAppCmd = &cobra.Command{
+		Use:   "app",
+		Short: "render Shipper Charts for an Application",
+		RunE:  renderChartFromApp,
+		PersistentPreRun: func(cmd *cobra.Command, args []string) {
+			if appName == "" && fileName == "" {
+				cmd.Printf("error: must have *one* of the flags: appName, filename\n")
+				os.Exit(1)
+			}
+			if appName != "" && fileName != "" {
+				cmd.Printf("error: must have *one* of the flags: appName, filename\n")
+				os.Exit(1)
+			}
+		},
+	}
+)
```

Comment:
I am not sure, but I wonder if there is a better way to not have these variables global?

Reply:
I can see it's consistent with the rest of the commands, but I was a bit confused by [this line](https://github.com/bookingcom/shipper/pull/374/files#diff-1a961418e54612e1a0d9ef44c8d799fbaca99d7551cdee04cd83e8b01560e784R60)

Reply:
The initial command was getting an appName, pulled it from the cluster and used the chart from it.
I wanted it to be possible to git a yaml of an application that does not exist in the cluster yet.
So with this command you can either give an app name or an application manifest (yaml file name)
The meaning of the variables is in their description in lines 40-41.
Other Go command line code is specifying the flags in a different location so there is no need for global variables. I opted for consistency, but I tend to agree that these global variables are not the best choice. I can start with these commands and have another PR to refactor the rest of shipperctl commands. WDYT?

Reply:
Thanks for the explanation, I think it makes sense to keep it like that for now. Would you consider this as a potential refactoring for the future?

Reply:
Of course! I wanted to do it by now but didn't have the time.
Go's way of doing CLIs is not so gracious, but I have seen some examples of nice ones when I was looking into this (`shipperctl decommission` shenanigans)